confusing error if step00-* are not executable

I found that if step00-* scripts are not executable, the CU log showed they couldn't be found (confusing error)

version 2.1.1

Here's a sample log entry

2017-03-14 11:24:24,037 [ERROR] log.py:17 - conjure-up/XXXXXXXXX: Unable to process step, missing /home/XXXXX/.cache/conjure-up/XXXXXXXXX/steps/step-01_example

This was worked around by chmod +x on the script.

--- a/conjureup/controllers/steps/gui.py
+++ b/conjureup/controllers/steps/gui.py
@@ -107,9 +107,14 @@ class StepsController:
             step_ex_path, ext = path.splitext(step_meta_path)
             if not path.isfile(step_ex_path) or \
                not os.access(step_ex_path, os.X_OK):
-                app.log.error(
-                    'Unable to process step, missing {}'.format(step_ex_path))
-                continue
+                failed_path = step_ex_path.split('/')[-3:]
+                msg = (
+                    'Step {} is not executable, make sure it has '
+                    'the executable bit set'.format(
+                        '/'.join(failed_path)))
+                app.log.error(msg)
+                self.__handle_exception('E002', Exception(msg))
+                return
             step_metadata = {}
             with open(step_meta_path) as fp:
                 step_metadata = yaml.load(fp.read())

--- a/conjureup/controllers/steps/tui.py
+++ b/conjureup/controllers/steps/tui.py
@@ -29,9 +29,14 @@ class StepsController:
             step_ex_path, ext = path.splitext(step_meta_path)
             if not path.isfile(step_ex_path) or \
                not os.access(step_ex_path, os.X_OK):
-                app.log.error(
-                    'Unable to process step, missing {}'.format(step_ex_path))
-                continue
+                failed_path = step_ex_path.split('/')[-3:]
+                msg = (
+                    'Step {} is not executable, make sure it has '
+                    'the executable bit set'.format(
+                        '/'.join(failed_path)))
+                app.log.error(msg)
+                utils.error(msg)
+                sys.exit(1)
             step_metadata = {}
             with open(step_meta_path) as fp:
                 step_metadata = yaml.load(fp.read())

